api/middleware/rateLimiter.go

```diff
@@ -0,0 +1,106 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/api/shared"
+	"github.com/gin-gonic/gin"
+)
+
+type rateLimiter struct {
+	requestsMap    map[string]uint64
+	mutRequestsMap sync.RWMutex
+	limits         map[string]uint64
+	mutLimits      sync.RWMutex
+	countDuration  uint64
```

Comment:
this could have been a time.Duration value and the print from L49 could have been:
```GO
printMessage := fmt.Sprintf("your IP exceeded the limit of %d requests in %v for this endpoint", limitForEndpoint, rl.countDuration)
```

Reply:
done